From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matti=20Lehtim=C3=A4ki?= <matti.lehtimaki@jolla.com>
Date: Sun, 11 May 2025 00:14:11 +0300
Subject: [PATCH] Revert "Prefer invoking methods by pointer rather than by
 name"

Reverted for Qt 5.6 compatibility.

This reverts commit 62cf686df1e9a647afe11825116a00043a971335.
---
 src/transactionprivate.cpp | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/src/transactionprivate.cpp b/src/transactionprivate.cpp
index 633e1bde7b657e18e9a4beb2b157c2bd0b1be480..6b2817c3a74629031f87d3c4c879fff75a9b84ee 100644
--- a/src/transactionprivate.cpp
+++ b/src/transactionprivate.cpp
@@ -297,40 +297,40 @@ void TransactionPrivate::updateProperties(const QVariantMap &properties)
         const QVariant &value = it.value();
         if (property == QLatin1String("AllowCancel")) {
             allowCancel = value.toBool();
-            QMetaObject::invokeMethod(q, &Transaction::allowCancelChanged, Qt::QueuedConnection);
+            QMetaObject::invokeMethod(q, "allowCancelChanged", Qt::QueuedConnection);
         } else if (property == QLatin1String("CallerActive")) {
             callerActive = value.toBool();
-            QMetaObject::invokeMethod(q, &Transaction::isCallerActiveChanged, Qt::QueuedConnection);
+            QMetaObject::invokeMethod(q, "isCallerActiveChanged", Qt::QueuedConnection);
         } else if (property == QLatin1String("DownloadSizeRemaining")) {
             downloadSizeRemaining = value.toLongLong();
-            QMetaObject::invokeMethod(q, &Transaction::downloadSizeRemainingChanged, Qt::QueuedConnection);
+            QMetaObject::invokeMethod(q, "downloadSizeRemainingChanged", Qt::QueuedConnection);
         } else if (property == QLatin1String("ElapsedTime")) {
             elapsedTime = value.toUInt();
-            QMetaObject::invokeMethod(q, &Transaction::elapsedTimeChanged, Qt::QueuedConnection);
+            QMetaObject::invokeMethod(q, "elapsedTimeChanged", Qt::QueuedConnection);
         } else if (property == QLatin1String("LastPackage")) {
             lastPackage = value.toString();
-            QMetaObject::invokeMethod(q, &Transaction::lastPackageChanged, Qt::QueuedConnection);
+            QMetaObject::invokeMethod(q, "lastPackageChanged", Qt::QueuedConnection);
         } else if (property == QLatin1String("Percentage")) {
             percentage = value.toUInt();
-            QMetaObject::invokeMethod(q, &Transaction::percentageChanged, Qt::QueuedConnection);
+            QMetaObject::invokeMethod(q, "percentageChanged", Qt::QueuedConnection);
         } else if (property == QLatin1String("RemainingTime")) {
             remainingTime = value.toUInt();
             q->remainingTimeChanged();
         } else if (property == QLatin1String("Role")) {
             role = static_cast<Transaction::Role>(value.toUInt());
-            QMetaObject::invokeMethod(q, &Transaction::roleChanged, Qt::QueuedConnection);
+            QMetaObject::invokeMethod(q, "roleChanged", Qt::QueuedConnection);
         } else if (property == QLatin1String("Speed")) {
             speed = value.toUInt();
-            QMetaObject::invokeMethod(q, &Transaction::speedChanged, Qt::QueuedConnection);
+            QMetaObject::invokeMethod(q, "speedChanged", Qt::QueuedConnection);
         } else if (property == QLatin1String("Status")) {
             status = static_cast<Transaction::Status>(value.toUInt());
-            QMetaObject::invokeMethod(q, &Transaction::statusChanged, Qt::QueuedConnection);
+            QMetaObject::invokeMethod(q, "statusChanged", Qt::QueuedConnection);
         } else if (property == QLatin1String("TransactionFlags")) {
             transactionFlags = static_cast<Transaction::TransactionFlags>(value.toUInt());
-            QMetaObject::invokeMethod(q, &Transaction::transactionFlagsChanged, Qt::QueuedConnection);
+            QMetaObject::invokeMethod(q, "transactionFlagsChanged", Qt::QueuedConnection);
         } else if (property == QLatin1String("Uid")) {
             uid = value.toUInt();
-            QMetaObject::invokeMethod(q, &Transaction::uidChanged, Qt::QueuedConnection);
+            QMetaObject::invokeMethod(q, "uidChanged", Qt::QueuedConnection);
         } else if (property == QLatin1String("Sender")) {
             senderName = value.toString();
             QMetaObject::invokeMethod(q, &Transaction::senderNameChanged, Qt::QueuedConnection);
